proxy-node-test/src/main/java/uk/gov/ida/notification/helpers/EidasAuthnRequestBuilder.java

```diff
@@ -48,7 +51,7 @@ public AuthnRequest build() throws TransformerException, SignatureException, Mar
         if (this.signingCredential != null) {
             SignatureBuilder signatureBuilder = SignatureBuilder
                     .aSignature()
-                    .withSignatureAlgorithm(new SignatureRSASHA256())
+                    .withSignatureAlgorithm(Optional.ofNullable(signatureAlgorithm).orElse(new SignatureRSASHA256()))
```

Comment:
I think the logic should be inside the definition of `withSignatureAlgorithm`
   ``` 
public EidasAuthnRequestBuilder withSignatureAlgorithm(SignatureAlgorithm signatureAlgorithm) {
        this.signatureAlgorithm = signatureAlgorithm == null ? new SignatureRSASHA256() :  signatureAlgorithm ;
        return this;
 }
```
I don't think we need optional here as well

Reply:
I understand why you've done this, I guess it is fine.

Reply:
May be this instead
` ``               .withSignatureAlgorithm(Optional.ofNullable(signatureAlgorithm).orElseGet(SignatureRSASHA256::new))
```

Reply:
Pushed a change for this